### PR TITLE
feat(chat): Interactive charts and options, expanded docs, fixed scrolling issue in slotted mode

### PR DESCRIPTION
**Pull request by ocornec**
Closes #

{{short description}}

#### Changelog

**New**

- {{new thing}}

**Changed**

- {{change thing}}

**Removed**

- {{removed thing}}

#### Testing / Reviewing

{{ Add descriptions, steps or a checklist for how reviewers can verify this PR works or not }}

### CODE REVIEW

File: packages/chat/components/chartElement/__stories__/chart.mdx [hunk 1]
```diff
@@ -46,13 +38,10 @@ import '@carbon-labs/ai-chat-chart/es/index.js';
 ## Implementation
 
 ### &lt;clabs-chat-chart&gt; attributes and properties
-
-<ArgTypes of="clabs-chat-chart" />
+<Controls story="Playground" />
```
Comment (annawen1) [inline packages/chat/components/chartElement/__stories__/chart.mdx:41]: Should we stick to `<ArgTypes of="clabs-chat-chart" />` here? With that, it displays all the attributes associated with chart component

File: packages/chat/components/chartElement/__stories__/chart.mdx [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { ArgTypes, Markdown, Meta } from '@storybook/blocks';
+import { ArgTypes, Markdown, Meta, Controls, ArgsTable } from '@storybook/blocks';
```
Comment (annawen1) [inline packages/chat/components/chartElement/__stories__/chart.mdx:1]: Don't think we would need `Controls` or `ArgsTable` if we're using `ArgTypes`

File: packages/chat/components/chartElement/__stories__/chart.mdx (hunk 1 above)
Comment (annawen1) [reply packages/chat/components/chartElement/__stories__/chart.mdx:41]: actually nvm, I see both `ArgTypes` and `Control` do similar things

File: packages/chat/components/chat/src/chat.ts [hunk 3]
```diff
@@ -187,7 +199,8 @@ export default class CLABSChat extends LitElement {
     const exampleMessageArray = this.sampleQuery.split('bot:');
     const userMessage = exampleMessageArray[0].replace('user:', '');
     const botMessage = exampleMessageArray[1];
-
+    console.log(userMessage);
+    console.log(botMessage);
```
Comment (annawen1) [inline packages/chat/components/chat/src/chat.ts:203]: Are these console logs needed?

File: packages/chat/components/chat/src/chat.ts [hunk 4]
```diff
@@ -197,6 +210,7 @@ export default class CLABSChat extends LitElement {
         index: 0,
       },
     ];
+    console.log(this._messages);
```
Comment (annawen1) [inline packages/chat/components/chat/src/chat.ts:213]: Is this console.log needed?

File: packages/chat/components/footer/src/footer.ts [hunk 5]
```diff
@@ -196,6 +234,14 @@ export default class footer extends LitElement {
...
   }
 
+  /**
+   * set focus on component when text area is focused
+   * @param {event} event - lit event sent by textarea focus
+   */
+  _textAreaIsFocused(event) {
+    console.log(event);
+  }
+
```
Comment (annawen1) [inline packages/chat/components/footer/src/footer.ts:244]: is this method needed?

File: packages/chat/components/footer/src/footer.ts [hunk 6]
```diff
@@ -160,7 +195,10 @@ export default class footer extends LitElement {
   /** handleMenuFileUpload - upload event in footer menu
    * @param {event} event - lit event sent by the file uploader in menu
    **/
-  _handleMenuFileUpload() {}
+  _handleMenuFileUpload(event) {
+    const files = event.detail?.addedFiles;
+    console.log(files[0]);
+  }
```
Comment (annawen1) [inline packages/chat/components/footer/src/footer.ts:201]: Is this method needed? Seems to only console log?

File: packages/chat/components/chartElement/__stories__/chart.mdx (hunk 1 above)
Comment (ocornec) [reply packages/chat/components/chartElement/__stories__/chart.mdx:41]: I did think showing the attributes was weird but didn't see a difference when changing, I'll try again and push

File: packages/chat/components/chartElement/__stories__/chart.mdx (hunk 2 above)
Comment (ocornec) [reply packages/chat/components/chartElement/__stories__/chart.mdx:1]: Alright!

File: packages/chat/components/chartElement/__stories__/chart.mdx (hunk 1 above)
Comment (annawen1) [inline packages/chat/components/chartElement/__stories__/chart.mdx:41]:
```suggestion
<ArgTypes of="clabs-chat-chart" />
```

**Review by annawen1 — APPROVED** [review]

**Pull request merged**